settings/Database.boot.setting.php

```diff
@@ -0,0 +1,124 @@
+<?php
+/*
+ +--------------------------------------------------------------------+
+ | Copyright CiviCRM LLC. All rights reserved.                        |
+ |                                                                    |
+ | This work is published under the GNU AGPLv3 license with some      |
+ | permitted exceptions and without any warranty. For full license    |
+ | and copyright information, see https://civicrm.org/licensing       |
+ +--------------------------------------------------------------------+
+ */
+
+/**
+ *
+ * @package CRM
+ * @copyright CiviCRM LLC https://civicrm.org/licensing
+ */
+
+/**
+ * Settings metadata file
+ */
+return [
+  'civicrm_db_dsn' => [
+    'name' => 'civicrm_db_dsn',
+    'title' => ts('CiviCRM Database Connection String'),
+    'description' => ts("The DSN for the CiviCRM Database."),
+    'group' => 'database',
+    'group_name' => 'CiviCRM Database',
+    'type' => 'String',
+    'html_type' => 'text',
+    //'default' => '',
+    'add' => '5.76',
+    'is_domain' => '1',
+    'is_contact' => 0,
+    'help_text' => NULL,
+    'is_constant' => TRUE,
+    'load_from_env' => TRUE,
+    'global_name' => 'CIVICRM_DSN',
```

Comment:
> (@ufundo ) edit civicrm.settings.php - instead of using the PHP constant set environment variables... And site still works...
>
> Potential gotcha at this stage: if you have `if (!defined('CIVICRM_DSN')) define('CIVICRM_DSN', );` in your civicrm.settings.php, this will jump in and set the DSN explicitly before the interpolation from components happens.

OK, cool. So riffing on this suggested experiment, I edited `CIVICRM_DSN` to specifically to put in *bad DSN* -- and then called Civi with an env-var containinga good DSN:

```bash
CIVICRM_DSN="mysql://s3civi_jq9sa:XXXX@127.0.0.1:3307/s3civi_jprre?new_link=true" \
  cv ev 'echo 123;'
```

And it worked with the env-var. 😄 

Similarly, I tried piece-wise credentials. (And per your  hint, I removed the `define()`.) 

```bash
CIVICRM_DB_USER=s3civi_jq9sa \
  CIVICRM_DB_PASSWORD=XXX \
  CIVICRM_DB_HOST=127.0.0.1 \
  CIVICRM_DB_PORT=3307 \
  CIVICRM_DB_NAME=s3civi_jprre  \
  cv ev 'echo 123;'
```

This also worked. 🎉 

(*All the above required the patch for `SettingsManager::bootSettings()` to work with cv.*)

Reply:
Hurrah :grinning: